generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs

```diff
@@ -0,0 +1,271 @@
+            {
+                if (activity.MembersAdded != null)
+                {
+                    // Iterate over all new members added to the conversation.
+                    foreach (var member in activity.MembersAdded)
+                    {
+                        // Greet anyone that was not the target (recipient) of this message.
+                        // To learn more about Adaptive Cards,
+                        // See https://aka.ms/msbot-adaptivecards for more details.
+                        if (member.Id != activity.Recipient.Id)
+                        {
+                            var welcomeCard = CreateAdaptiveCardAttachment();
+                            var response = CreateResponse(activity, welcomeCard);
+                            await dc.Context.SendActivityAsync(response);
+                        }
+                    }
+                }
+            }
+
+            await _conversationState.SaveChangesAsync(turnContext);
+            await _userState.SaveChangesAsync(turnContext);
+        }
+
+        // Determine if an interruption has occurred before we dispatch to any active dialog.
+        private async Task<bool> IsTurnInterruptedAsync(DialogContext dc, string topIntent)
+        {
+            // See if there are any conversation interrupts we need to handle.
+            if (topIntent.Equals(CancelIntent))
+            {
+                if (dc.ActiveDialog != null)
+                {
+                    await dc.CancelAllDialogsAsync();
+                    await dc.Context.SendActivityAsync("Ok. I've canceled our last activity.");
+                }
+                else
+                {
+                    await dc.Context.SendActivityAsync("I don't have anything to cancel.");
+                }
+
+                return true;        // Handled the interrupt.
+            }
+
+            if (topIntent.Equals(HelpIntent))
+            {
+                await dc.Context.SendActivityAsync("Let me try to provide some help.");
+                await dc.Context.SendActivityAsync("I understand greetings, being asked for help, or being asked to cancel what I am doing.");
+                if (dc.ActiveDialog != null)
+                {
+                    await dc.RepromptDialogAsync();
+                }
+
+                return true;        // Handled the interrupt.
+            }
+
+            return false;           // Did not handle the interrupt.
+        }
+
+        // Create an attachment message response.
+        private Activity CreateResponse(Activity activity, Attachment attachment)
+        {
+            var response = activity.CreateReply();
+            response.Attachments = new List<Attachment>() { attachment };
+            return response;
+        }
+
+        // Load attachment from file.
+        private Attachment CreateAdaptiveCardAttachment()
+        {
+            var adaptiveCard = File.ReadAllText(@".\Dialogs\Welcome\Resources\welcomeCard.json");
```

Comment:
This code needs to change to use Path.Combine which is environment agnostic.  I just fixed this in the samples/csharp_dotnet/13.basic-bot sample.  Please use: 
            string[] paths = {".", "Dialogs", "Welcome", "Resources", "welcomeCard.json"};
            string fullPath = Path.Combine(paths);
            var adaptiveCard = File.ReadAllText(fullPath);

Reply:
Feedback Applied